### ISSUE TITLE
Azure: Upload of decompressed FCOS image is very slow

### ISSUE THREAD

**Issue by jomeier** [openshift/okd#48]
Hi,

I couldn't manage to get the new Azure installer working because the upload of the decompressed (fcos image is xz compressed) takes so long (more than 30 Minutes on my PC) that this procedure seems not to be feasable to me.

Please support a procedure where users can upload the fcos image on their own to a storage account which is in a different resource group and will be referenced from all cluster VMs. 

A method to configure this storage blob container is necessary (through env variables which are fed to the installer?).

The time for the upload would be necessary only once for all OKD clusters and must not be spent for each cluster again and again.

Best regards,

Josef

**Comment by vrutkovs** [comment]
I am not familiar with Azure, what's the recommendation here? 

If Azure doesn't give us tools to upload the image faster there isn't much we can do about it

**Comment by jomeier** [comment]
The problem is in my opinion, that it takes almost half an hour every time the image is uploaded before the VMs get started. Every time we create a cluster.

The upload is limited by the upload speed of your provider.

We could ...

create a VM in Azure which downloads, decompressed and uploads the image to the blob storage. Because we still are on Azure everything works much faster.

Upload the image independent of the installer in a dedicated resource group and use it for multiple clusters.

Get the Image in the Azure Marketplace or an image gallery.

**Comment by jomeier** [comment]
https://github.com/openshift/installer/pull/3033

**Comment by sanderkooger** [comment]
Just tried to deploy a cluster to azure, Im running into the same problems.

**Comment by jomeier** [comment]
@sanderkooger:
In my video I explain how you can modify the openshift installer to get OKD 4.4 installed on Azure:

https://www.youtube.com/watch?v=PXZn87FCIYk

Have fun (and don't forget to like my video if it helped you :-) ).

**Comment by jomeier** [comment]
It's on minute 14:13

**Comment by cgwalters** [comment]
One generic thing you can do is just boot up any Linux machine in Azure and run `openshift-install` from there over ssh or whatever.

(There's also https://github.com/openshift/hive/ once you have one cluster running)

**Comment by sanderkooger** [comment]
I just created a fresh Ubuntu box and tried from within azure. No joy... 

```
DEBUG azurerm_storage_blob.rhcos_image: Still creating... [50s elapsed]
DEBUG azurerm_storage_blob.rhcos_image: Still creating... [1m0s elapsed]
DEBUG azurerm_storage_blob.rhcos_image: Still creating... [1m10s elapsed]
DEBUG azurerm_storage_blob.rhcos_image: Still creating... [1m20s elapsed]
DEBUG azurerm_storage_blob.rhcos_image: Still creating... [1m30s elapsed]
DEBUG azurerm_storage_blob.rhcos_image: Creation complete after 1m31s [id=https://cluster3iydm.blob.core.windows.net/vhd/rhcos3iydm.vhd]
DEBUG azurerm_image.cluster: Creating...
ERROR
ERROR Warning: "resource_group_name": [DEPRECATED] This field has been deprecated and is no longer used - will be removed in 2.0 of the Azure Provider
ERROR
ERROR   on ../../tmp/openshift-install-362141893/main.tf line 163, in resource "azurerm_storage_container" "vhd":
ERROR  163: resource "azurerm_storage_container" "vhd" {
ERROR
ERROR (and 3 more similar warnings elsewhere)
ERROR
ERROR
ERROR Error: compute.ImagesClient#CreateOrUpdate: Failure sending request: StatusCode=400 -- Original Error: Code="InvalidParameter" Message="The source blob https://cluster3iydm.blob.core.windows.net/vhd/rhcos3iydm.vhd is not a page blob." Target="disks"
ERROR
ERROR   on ../../tmp/openshift-install-362141893/main.tf line 179, in resource "azurerm_image" "cluster":
ERROR  179: resource "azurerm_image" "cluster" {
ERROR
ERROR
FATAL failed to fetch Cluster: failed to generate asset "Cluster": failed to create cluster: failed to apply using Terraform
```

Its my first cluster so hyve is not an option.

**Comment by sanderkooger** [comment]
@jomeier I'm trying your YouTube vid but on cherry-pick item give me an error.

```
git cherry-pick 05a7dff57a68134a5457e98864e62783221a5aa
fatal: bad revision '05a7dff57a68134a5457e98864e62783221a5aa'
```

**Comment by jomeier** [comment]
@sanderkooger:
Hm ... I just tried it out again and on my site it works (git version 2.17.1)

**Comment by jomeier** [comment]
@sanderkooger  
I reported this "is not a page blob" several times. My hack also takes care about that.

**Comment by sanderkooger** [comment]
@jomeier Thats weird as I can cherrypick the other commits. Its just the last one i cant pick.

**Comment by jomeier** [comment]
@sanderkooger 
You can leave that one away. With this commit the installation will do a little bit faster but it should work without it for the moment.

**Comment by sanderkooger** [comment]
let me retry

**Comment by jomeier** [comment]
@sanderkooger 
Could you try out this and confirm that it works, please?

**Comment by sanderkooger** [comment]
@jomeier Is there a place where we can chat privately? Answered in your new issue. Having issues using your YouTube as i have to copy the identifiers for the commits by hand.  Mail me at sander@thisisfashion.tv please.

**Comment by jomeier** [comment]
@sanderkooger 
Come in the kubernetes/#openshift-dev channel.

**Comment by jomeier** [comment]
If you put the latest nightly in an Azure VM or Cloud Shell, the Installation will succeed without further patches.

**Comment by sanderkooger** [comment]
Maybe ill try, I actually have been doing the math, The infrastructure is meant for a small startup ( unfunded) and I kind of figured that maybe its better to run AKS instead of openshift, It’s a little less developer friendly in terms of automates stream2images, but on the other hand During development I only have to run one workernode and a mysql db.

Running 2 4core masternodes, 2 4core infranodes, and the workernodes gets a little too expensive too fast.

Thank you for the support though! And good work!
